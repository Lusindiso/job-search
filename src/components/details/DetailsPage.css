.details-headline-container {
  display: flex;
  justify-content: center;
  background-color: var(--color-primary-400);
}

.details-headline {
  position: relative;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: var(--headline-height);
  width: 100%;
  max-width: var(--width-max);
  padding: 0 20px;
}

.details-headline-bg {
  position: absolute;
  left: 25px;
  font-family: 'Crimson Text', serif;
  font-size: 100px;
  font-weight: 600;
  color: var(--color-primary-900);
  text-transform: lowercase;
}

.details-headline-group {
  position: relative;
  width: 60%;
  text-align: end;
}

.details-headline-keywords {
  text-transform: uppercase;
  font-size: 18px;
  font-weight: 600;
}

.details-cards-section {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.details-card-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: var(--color-primary-600);
  width: 100%;
  min-height: 120px;
  max-width: var(--width-max);
  padding: 15px;
  border: none;
  text-align: start;
}

.details-card-container:nth-child(even) {
  background-color: var(--color-primary-800);
}

.job-info1 {
  width: 100%;
  display: flex;
  justify-content: space-between;
  gap: 15px;
}

.job-info1-left {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.job-info1 .icon-arrow-right {
  font-size: 20px;
  font-weight: 300;
}

.job-info1-employer {
  font-size: 14px;
}

.job-info1-title {
  font-size: 18px;
  font-weight: 600;
}

.job-info2 {
  width: 100%;
  display: grid;
  grid-template-columns: 3fr 1fr;
  font-size: 13px;
}

.job-info2-day {
  text-align: end;
}
