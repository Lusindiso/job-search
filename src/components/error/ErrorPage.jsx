const ErrorPage = () => (
  <div>
    404: Not found
  </div>
);

export default ErrorPage;
