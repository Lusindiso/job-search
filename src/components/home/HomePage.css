.homepage {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.headline {
  width: 100%;
  height: var(--headline-height);
  padding-top: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--color-primary-400);
  background-image: url('../../../public/images/capstone3-rocket-01.svg');
  background-size: contain;
  background-origin: content-box;
  background-position: bottom;
  background-repeat: no-repeat;
}

.headline form {
  display: flex;
  justify-content: space-between;
  gap: 15px;
  width: 80%;
  max-width: 800px;
  margin-top: -40px;
}

.headline input {
  width: 100%;
  padding: 10px 13px;
  border: none;
  border-radius: 5px;
  color: var(--color-grey-800);
  outline: none;
}

.headline button {
  padding: 10px 13px;
  background-color: var(--color-primary-900);
  border: none;
  border-radius: 5px;
}

.cards-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 100%;
  max-width: 800px;
}

.card-container {
  position: relative;
  height: 187.5px;
  border: none;
  background-color: var(--color-primary-600);
  overflow: hidden;
  padding: 10px 10px 3px 10px;
}

.card-container:nth-child(4n-1),
.card-container:nth-child(4n-2) {
  background-color: var(--color-primary-800);
}

.card-bg {
  position: absolute;
  left: 25px;
  font-family: 'Crimson Text', serif;
  font-size: 100px;
  font-weight: 600;
  color: var(--color-primary-900);
  text-transform: lowercase;
}

.card {
  position: relative;
  display: grid;
  grid-template-rows: 1fr 4fr 1fr;
  justify-items: end;
  height: 100%;
}

.card .icon-arrow-right {
  font-size: 30px;
  font-weight: 300;
}

.card-title {
  align-self: end;
  text-transform: uppercase;
  font-size: 18px;
  font-weight: 600;
  text-align: end;
}

.card-numberofjobs {
  font-size: 15px;
  font-weight: 400;
}
