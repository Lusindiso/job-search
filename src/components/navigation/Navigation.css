.nav {
  display: grid;
  grid-template-columns: 1fr auto 1fr;
  background-color: var(--color-primary-500);
  height: 50px;
}

.nav-title {
  text-align: center;
  font-weight: 600;
  margin: auto;
}

.nav-button {
  display: flex;
  align-items: center;
  padding: 0 10px;
  background-color: transparent;
  border: none;
}
