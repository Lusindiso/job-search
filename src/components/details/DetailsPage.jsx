import { useRef } from 'react';
import { useSelector } from 'react-redux';
import DetailsCard from './sub_components/DetailsCard';
import './DetailsPage.css';

const DetailsPage = () => {
  const data = useSelector((state) => state.jobsReducer);
  const page = useRef(0);

  return (
    <div>
      <section className="details-headline-container">
        <div className="details-headline">
          <div className="details-headline-bg">{data?.keywords?.match(/\b(\w)/g)}</div>
          <div className="details-headline-group">
            <div className="details-headline-keywords">{data?.keywords}</div>
            <div>
              {data?.totalResults}
              &nbsp;jobs found
            </div>
          </div>
        </div>
      </section>
      <div className="section-title-container">
        <div className="section-title">SEARCH RESULT</div>
      </div>
      <section className="details-cards-section">
        {data?.results?.slice(page.current, page.current + 50)
          .map((s) => (
            <DetailsCard
              key={s.jobId}
              jobTitle={s.jobTitle}
              employerName={s.employerName}
              locationName={s.locationName}
              minimumSalary={s.minimumSalary || 0}
              maximumSalary={s.maximumSalary || 0}
              currency={s.currency || ''}
              date={s.date}
            />
          ))}
      </section>
    </div>
  );
};

export default DetailsPage;
