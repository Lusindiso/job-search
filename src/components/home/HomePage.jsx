import { useEffect, useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { aggregateFetch } from '../../redux/aggregate/aggregate';
import Headline from './sub_components/Headline';
import Card from './sub_components/Card';
import './HomePage.css';

const HomePage = () => {
  const data = useSelector((state) => state.aggregateReducer);
  const dispatch = useDispatch();
  const categories = ['frontend', 'frontend remote', 'react developer', 'fullstack', 'fullstack remote', 'ruby developer'];
  // A helper variable to keep track of the number of categories that has been fetched
  const categoriesTracker = useRef(data.length);
  // When data gets updated, fetch another one until all categories are fetched
  useEffect(() => {
    if (categoriesTracker.current < categories.length) {
      dispatch(aggregateFetch(categories[categoriesTracker.current]));
      categoriesTracker.current += 1;
    }
  }, [data]);

  return (
    <div className="homepage">
      <Headline />
      <div className="section-title-container">
        <div className="section-title">OUR TOP PICKS</div>
      </div>
      <section className="cards-container">
        {categories.map((category) => (
          <Card
            key={category}
            cardTitle={category}
            numberOfJobs={data.filter((d) => d.keywords === category)[0]?.totalResults || 0}
          />
        ))}
      </section>
    </div>
  );
};

export default HomePage;
