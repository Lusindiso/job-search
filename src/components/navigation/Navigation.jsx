import { useDispatch } from 'react-redux';
import { useLocation, useNavigate } from 'react-router-dom';
import { clearData } from '../../redux/jobs/jobs';
import './Navigation.css';

const Navigation = () => {
  const location = useLocation();
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const handleBack = () => {
    dispatch(clearData());
    navigate('/');
  };

  return (
    <nav className="nav">
      {location.pathname === '/'
        ? <div />
        : (
          <button className="nav-button" type="button" onClick={handleBack}>
            <span className="material-symbols-outlined">arrow_back_ios_new</span>
          </button>
        )}
      {location.pathname === '/'
        ? <div className="nav-title font-lato">Reed Job Search</div>
        : <div className="nav-title font-lato">Search results</div>}
    </nav>
  );
};

export default Navigation;
