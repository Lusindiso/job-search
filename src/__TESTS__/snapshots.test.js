import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import HomePage from '../components/home/HomePage';
import DetailsPage from '../components/details/DetailsPage';
import ErrorPage from '../components/error/ErrorPage';
import Navigation from '../components/navigation/Navigation';

describe('Snapshot test for Home page', () => {
  it('It should render an identical design', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <HomePage />
        </Router>
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Snapshot test for Details page', () => {
  it('It should render an identical design', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <DetailsPage />
        </Router>
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Snapshot test for Error page', () => {
  it('It should render an identical design', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <ErrorPage />
        </Router>
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Snapshot test for Navigaion page', () => {
  it('It should render an identical design', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Navigation />
        </Router>
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
