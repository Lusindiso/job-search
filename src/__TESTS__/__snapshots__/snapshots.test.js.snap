// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Details page It should render an identical design 1`] = `
<div>
  <section
    className="details-headline-container"
  >
    <div
      className="details-headline"
    >
      <div
        className="details-headline-bg"
      />
      <div
        className="details-headline-group"
      >
        <div
          className="details-headline-keywords"
        />
        <div>
           jobs found
        </div>
      </div>
    </div>
  </section>
  <div
    className="section-title-container"
  >
    <div
      className="section-title"
    >
      SEARCH RESULT
    </div>
  </div>
  <section
    className="details-cards-section"
  />
</div>
`;

exports[`Snapshot test for Error page It should render an identical design 1`] = `
<div>
  404: Not found
</div>
`;

exports[`Snapshot test for Home page It should render an identical design 1`] = `
<div
  className="homepage"
>
  <section
    className="headline"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        maxLength="50"
        placeholder="e.g. frontend react remote"
        required={true}
        type="text"
      />
      <button
        type="submit"
      >
        SEARCH
      </button>
    </form>
  </section>
  <div
    className="section-title-container"
  >
    <div
      className="section-title"
    >
      OUR TOP PICKS
    </div>
  </div>
  <section
    className="cards-container"
  >
    <button
      className="card-container"
      onClick={[Function]}
      type="button"
    >
      <div
        className="card-bg"
      >
        f
      </div>
      <div
        className="card"
      >
        <span
          className="material-symbols-outlined icon-arrow-right"
        >
          arrow_circle_right
        </span>
        <div
          className="card-title"
        >
          frontend
        </div>
        <div
          className="card-numberofjobs"
        >
          0
           jobs
        </div>
      </div>
    </button>
    <button
      className="card-container"
      onClick={[Function]}
      type="button"
    >
      <div
        className="card-bg"
      >
        f
        r
      </div>
      <div
        className="card"
      >
        <span
          className="material-symbols-outlined icon-arrow-right"
        >
          arrow_circle_right
        </span>
        <div
          className="card-title"
        >
          frontend remote
        </div>
        <div
          className="card-numberofjobs"
        >
          0
           jobs
        </div>
      </div>
    </button>
    <button
      className="card-container"
      onClick={[Function]}
      type="button"
    >
      <div
        className="card-bg"
      >
        r
        d
      </div>
      <div
        className="card"
      >
        <span
          className="material-symbols-outlined icon-arrow-right"
        >
          arrow_circle_right
        </span>
        <div
          className="card-title"
        >
          react developer
        </div>
        <div
          className="card-numberofjobs"
        >
          0
           jobs
        </div>
      </div>
    </button>
    <button
      className="card-container"
      onClick={[Function]}
      type="button"
    >
      <div
        className="card-bg"
      >
        f
      </div>
      <div
        className="card"
      >
        <span
          className="material-symbols-outlined icon-arrow-right"
        >
          arrow_circle_right
        </span>
        <div
          className="card-title"
        >
          fullstack
        </div>
        <div
          className="card-numberofjobs"
        >
          0
           jobs
        </div>
      </div>
    </button>
    <button
      className="card-container"
      onClick={[Function]}
      type="button"
    >
      <div
        className="card-bg"
      >
        f
        r
      </div>
      <div
        className="card"
      >
        <span
          className="material-symbols-outlined icon-arrow-right"
        >
          arrow_circle_right
        </span>
        <div
          className="card-title"
        >
          fullstack remote
        </div>
        <div
          className="card-numberofjobs"
        >
          0
           jobs
        </div>
      </div>
    </button>
    <button
      className="card-container"
      onClick={[Function]}
      type="button"
    >
      <div
        className="card-bg"
      >
        r
        d
      </div>
      <div
        className="card"
      >
        <span
          className="material-symbols-outlined icon-arrow-right"
        >
          arrow_circle_right
        </span>
        <div
          className="card-title"
        >
          ruby developer
        </div>
        <div
          className="card-numberofjobs"
        >
          0
           jobs
        </div>
      </div>
    </button>
  </section>
</div>
`;

exports[`Snapshot test for Navigaion page It should render an identical design 1`] = `
<nav
  className="nav"
>
  <div />
  <div
    className="nav-title font-lato"
  >
    Reed Job Search
  </div>
</nav>
`;
